src/main/java/uk/gov/companieshouse/confirmationstatementapi/interceptor/SubmissionInterceptor.java

```diff
@@ -32,18 +32,19 @@ public boolean preHandle(HttpServletRequest request, HttpServletResponse respons
         final Map<String, String> pathVariables = (Map<String, String>) request
                 .getAttribute(HandlerMapping.URI_TEMPLATE_VARIABLES_ATTRIBUTE);
         final var submissionId = pathVariables.get(CONFIRMATION_STATEMENT_ID_KEY);
+        final var transactionId = pathVariables.get(TRANSACTION_ID_KEY);
 
         var logMap = new HashMap<String, Object>();
         String reqId = request.getHeader(ERIC_REQUEST_ID_KEY);
 
         Optional<ConfirmationStatementSubmissionJson> submission = confirmationStatementService.getConfirmationStatement(submissionId);
         boolean isPresent = submission.isPresent();
-        if (isPresent) {
-            ApiLogger.infoContext(reqId, "Submission found for " + submissionId, logMap);
+
+        if (isPresent && submission.get().getLinks().get("self").contains(transactionId)) {
```

Comment:
I think we should still add the submission into the request even if its self link doesn't have a transaction id. we wouldn't want to say not found if it is found but the self link isnt right.